src/app/modules/spcp/sp.oidc.service.ts

```diff
@@ -0,0 +1,310 @@
+import fs from 'fs'
+import { err, errAsync, ok, okAsync, Result, ResultAsync } from 'neverthrow'
+
+import {
+  ISpcpMyInfo,
+  spcpMyInfoConfig,
+} from '../../config/features/spcp-myinfo.config'
+import { createLoggerWithLabel } from '../../config/logger'
+
+import { SpOidcClient } from './sp.oidc.client'
+import { PublicJwks } from './sp.oidc.client.types'
+import {
+  CreateJwtError,
+  CreateRedirectUrlError,
+  ExchangeAuthTokenError,
+  InvalidIdTokenError,
+  InvalidJwtError,
+  InvalidStateError,
+  MissingJwtError,
+  VerifyJwtError,
+} from './spcp.errors'
+import {
+  JwtName,
+  JwtPayload,
+  JwtPayloadFromCookie,
+  ParsedSpcpParams,
+  SingpassJwtPayloadFromCookie,
+  SpcpCookies,
+  SpcpDomainSettings,
+} from './spcp.types'
+import { extractFormId, isSingpassJwtPayload } from './spcp.util'
+
+const logger = createLoggerWithLabel(module)
+
+/**
+ * Class for executing Singpass OIDC-related services.
+ * Exported for testing.
+ */
+export class SpOidcServiceClass {
+  #spOidcProps: ISpcpMyInfo
+  #spOidcClient: SpOidcClient
+
+  constructor(props: ISpcpMyInfo) {
+    this.#spOidcClient = new SpOidcClient({
+      spOidcRpClientId: props.spOidcRpClientId,
+      spOidcRpRedirectUrl: props.spOidcRpRedirectUrl,
+      spOidcNdiDiscoveryEndpoint: props.spOidcNdiDiscoveryEndpoint,
+      spOidcNdiJwksEndpoint: props.spOidcNdiJwksEndpoint,
+      spOidcRpPublicJwks: JSON.parse(
+        fs.readFileSync(props.spOidcRpJwksPublicPath).toString(),
+      ),
+      spOidcRpSecretJwks: JSON.parse(
+        fs.readFileSync(props.spOidcRpJwksSecretPath).toString(),
+      ),
```

Comment:
these functions may throw, will they be caught if so?

Reply:
No - if the functions throw (e.g. file does not exist at the specified path), the `SpOidcServiceClass` constructor will throw an error, the app will crash and the instances will be terminated (before they are deployed to accept traffic). I think this is better than trying to catch and handle the error because there is no way that the class can be instantiated without a valid jwks.

Is also similar to the existing SpcpServiceClass.